pkg/network/aci_network.go

```diff
@@ -128,23 +128,27 @@ func (pn *ProviderNetwork) setupNetwork(ctx context.Context, azConfig *auth.Conf
 	var rawResponse *http.Response
 	ctxWithResp := runtime.WithCaptureResponse(ctx, &rawResponse)
 
-	createSubnet := true
+	updateSubnet := false
 	currentSubnet, err := pn.GetACISubnet(ctxWithResp, subnetsClient)
 	if err != nil {
 		return err
 	}
 
-	if err == nil {
-		createSubnet, err = pn.shouldCreateSubnet(currentSubnet, createSubnet)
+	createNewSubnet := false
+	if currentSubnet == (aznetworkv2.Subnet{}) {
+		createNewSubnet = true
+	} else {
+		updateSubnet = true
```

Comment:
You don't need this line, I believe. line 143 should give a correct value.

Reply:
updateSubnet needs to start out true because shouldCreateSubnet only toggles to false but I like [this suggestion](https://github.com/virtual-kubelet/azure-aci/pull/628#discussion_r1699259633) for readability that's why I set the updateSubnet to true here.

Reply:
updated